File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/ui/rate/RateResult.java
package pl.edu.amu.wmi.erykandroidcommon.ui.rate;

public class RateResult<T> {

    private static final String SUCCESS_STATUS = "updated";
    private T item;
    private String status;

    public RateResult() {
        // retrofit need
    }

    public RateResult(T item) {
        this.item = item;
        this.status = SUCCESS_STATUS;

    }

    public T getItem() {
        return item;
    }

    public boolean isSuccess() {
        return SUCCESS_STATUS.equals(status);
    }

}

File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/ui/rate/RateResult.kt
package pl.edu.amu.wmi.erykandroidcommon.ui.rate

internal class RateResult<T> {

    val item: T

    private val status: String

    val isSuccess: Boolean
        get() = SUCCESS_STATUS == status

    constructor() {
        // retrofit need
    }

    constructor(item: T) {
        this.item = item
        this.status = SUCCESS_STATUS

    }

    companion object {

        private val SUCCESS_STATUS = "updated"
    }

}
